src/ILCompiler.RyuJit/src/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -907,7 +907,7 @@ private CorInfoHelpFunc getCastingHelper(ref CORINFO_RESOLVED_TOKEN pResolvedTok
             return helper;
         }
 
-        private CorInfoHelpFunc getNewHelper(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_METHOD_STRUCT_* callerHandle)
+        private CorInfoHelpFunc getNewHelper(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_METHOD_STRUCT_* callerHandle, ref bool pHasSideEffects)
```

Comment:
Do we need to set `pHasSideEffects` to something? It is out arg.

Reply:
I altered the jit / R2R implementations to ensure if `pHasSideEffects != null` we initialize it based on whether the type has a finalizer.